extensions/amp-list/0.1/amp-list.js

```diff
@@ -467,46 +468,43 @@ export class AmpList extends AMP.BaseElement {
       this.container_.dispatchEvent(event);
 
       // Attempt to resize to fit new rendered contents.
-      this.attemptToFit_(this.container_, () => {
-        // If auto-resize is set, then change to container layout instead of
-        // changing height (with one exception).
-        if (this.element.hasAttribute('auto-resize')) {
-          const layout = this.element.getAttribute('layout');
-          if (layout == Layout.FLEX_ITEM) {
-            // TODO(cathyxz, #17824): Flex-item + reset-on-refresh will add
-            // an invisible loader that fills the amp-list and shoves all
-            // list items out of the amp-list.
-            return true;
-          } else if (layout !== Layout.CONTAINER) {
-            this.changeToLayoutContainer_(layout);
-          }
-          return false;
-        }
-        return true;
-      });
+      this.attemptToFit_(this.container_);
+
+      if (this.element.hasAttribute('auto-resize')) {
+        // If the element's size was changed, change to container layout
+        // if the auto-resize attribute is set.
+        this.element.signals().whenSignal(CommonSignals.SIZE_CHANGED)
+            .then(() => this.applyAutoResize_());
+      }
     });
   }
 
+  /**
+   * If the element has the auto-resize attribute, change to layout container.
+   * @private
+   */
+  applyAutoResize_() {
+    const layout = this.element.getAttribute('layout');
+    if (layout !== Layout.CONTAINER) {
```

Comment:
this implies that amp-list could have layout=container which is not the case and creates confusion around why we have `changeToLayoutContainer_` in the first place, although harmless, I suggest removing.

Reply:
The reason why this check exists is to avoid re-applying the changeToContainer logic if it has already been done. Maybe I can add a comment to that effect to clarify.